File: uva10920/uva10920.py
# 10920 - Spiral Tap
# Autores: Jackson Willian

import sys

def SpiralTap(ordem, numero):
	if (ordem % 2 != 0):
		meio = int(ordem / 2)
		if (numero == 1):
			return meio+1, meio+1
		elif (numero == (ordem**2) ):
			return ordem, ordem
		else:
			# submatriz comeca em ordem 3
			# anel , min , max, ordem, 
			anel = False
			subMatriz = [1, 2, 9, 3]
			while (anel == False) and ( subMatriz[2] < ordem**2):
				if (numero >= subMatriz[1]) and ( numero <= subMatriz[2] ): 
					anel = True
				else:
					subMatriz[0] += 1
					subMatriz[1] = subMatriz[2] + 1
					subMatriz[3] = subMatriz[3] + 2
					subMatriz[2] = subMatriz[3]**2

			linhaAcima = meio + subMatriz[0]
			colunaEsquerda = meio - subMatriz[0]
			linhaAbaixo = meio - subMatriz[0] 
			colunaDireita = meio + subMatriz[0]


			achou = False
			valorMaximo = subMatriz[2] # do anel
			espaco = subMatriz[3] - 1 # valor do quanto deve ser subtraido para encontrar o valor que está na linha

			intervalor1 = [ valorMaximo - espaco, valorMaximo]
			intervalor2 = [ valorMaximo - (espaco * 2), valorMaximo - espaco]
			intervalor3 = [ valorMaximo - (espaco * 3) , valorMaximo - (espaco * 2)]
			intervalor4 = [ valorMaximo - (espaco * 4) , valorMaximo - (espaco * 3)]

			if( numero >=  intervalor1[0] and numero <= intervalor1[1] ):
				valor = intervalor1[1]
				linhaAcima = linhaAcima - (valor - numero)
				return linhaAcima+1, colunaDireita+1
			elif( numero >=  intervalor2[0] and numero <= intervalor2[1] ):
				valor = intervalor2[1]
				colunaDireita = colunaDireita - (valor - numero)
				return linhaAbaixo+1, colunaDireita+1
			elif( numero >=  intervalor3[0] and numero <= intervalor3[1] ):
				valor = intervalor3[1]
				linhaAbaixo = linhaAbaixo + (valor- numero)
				return linhaAbaixo+1 , colunaEsquerda+1		
			elif ( numero >=  intervalor4[0] and numero <= intervalor4[1] ):
				valor = intervalor4[1]
				colunaEsquerda =  colunaEsquerda + (valor - numero)
				return linhaAcima+1, colunaEsquerda+1


def main():

	linha = sys.stdin.readline()
	while(linha != ""):
		ordem = int( str(linha.split()[0]) )
		numero = int( str(linha.split()[1]) )
		if(ordem !=0 and numero != 0):
			lin, col = SpiralTap(ordem, numero)
			print("Line = {}, column = {}.".format(lin, col))
		linha = sys.stdin.readline()
	
	return 0

main()

File: uva10920/uva10920.java
/* Main.java
 * UVa 10920 - Spiral Tap
 * Autores: Jackson Willian
 */

import java.util.ArrayList;
import java.util.Collections;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.lang.Math; 

class Main {

    public static void main(String[] args) {
        BufferedReader br;
        String linha, resposta;
        String[] vetorNumeros;
        long ordem, numero;

        br = new BufferedReader(new InputStreamReader(System.in));

        try {

            linha =  br.readLine();
            while( linha != null ) {
                vetorNumeros  = linha.split(" ");
                ordem = Long.parseLong(vetorNumeros[0]);
                numero = Long.parseLong(vetorNumeros[1]);;

                if(ordem != 0 && numero != 0){
                    
                     spiralTap(ordem, numero);
                
                }
                linha =  br.readLine();
            }
        }
        catch (Exception  e){
            System.out.println(e);
        }
      
    }

    
    public static void spiralTap ( long ordem, long numero){
        long meio = 0, linhaCima = 0, linhaAbaixo = 0, colunaEsquerda = 0, colunaDireita = 0;
        long camada = 0, minimo = 0, maximo = 0, subOrdem = 0;
        long valor = 0, linha = 0, coluna = 0, interGeral = 0, intervaloInicio = 0, intervaloFinal = 0;

        if(ordem % 2 != 0){
            meio = (long) (ordem / 2);
            if(numero == 1){
                linha = meio + 1;
                coluna = meio + 1;
                System.out.println("Line = "+linha+", column = "+coluna+".");
            }else if (numero == (long) Math.pow(ordem, 2)){
                linha = ordem;
                coluna = ordem;
                System.out.println("Line = "+linha+", column = "+coluna+".");               
            }else{

                camada = 1;
                minimo = 2;
                maximo = 9;
                subOrdem = 3;

                while( true ){

                    if(numero >= minimo && numero <= maximo){
                        break;
                    }

                    camada += 1;
                    subOrdem  += 2;
                    minimo = maximo + 1;
                    maximo = (long) Math.pow(subOrdem, 2);

                }

                linhaCima = meio + camada;                                                                                                                                                                                           
                colunaEsquerda = meio - camada;
                linhaAbaixo = meio - camada;
                colunaDireita = meio + camada;

                interGeral = subOrdem - 1;

                intervaloInicio = maximo - interGeral;
                intervaloFinal = (long) Math.pow(subOrdem, 2);
                if (numero >= intervaloInicio && numero <= intervaloFinal){
                    valor = intervaloFinal;
                    
                    linhaCima = linhaCima - (valor - numero);
                    linha = linhaCima+1;
                    coluna = colunaDireita+1;
                    System.out.println("Line = "+linha+", column = "+coluna+".");                     

                }else {
                    intervaloInicio = maximo - (interGeral * 2);
                    intervaloFinal = maximo - interGeral;
                if (numero >= intervaloInicio && numero <= intervaloFinal){
                        valor = intervaloFinal;
                        colunaDireita = colunaDireita - (valor - numero);
                        linha = linhaAbaixo+1;
                        coluna = colunaDireita+1;
                        System.out.println("Line = "+linha+", column = "+coluna+".");                          
                    }else{
                        intervaloInicio = maximo - (interGeral * 3);
                        intervaloFinal = maximo - (interGeral * 2);
                        if (numero >= intervaloInicio && numero <= intervaloFinal){
                            valor = intervaloFinal;
                            linhaAbaixo = linhaAbaixo + (valor - numero);
                            linha = linhaAbaixo+1;
                            coluna = colunaEsquerda+1;
                            System.out.println("Line = "+linha+", column = "+coluna+".");  

                        }else{
                            intervaloInicio = maximo - (interGeral * 4);
                            intervaloFinal = maximo - (interGeral * 3);
                            if(numero >= intervaloInicio && numero <= intervaloFinal){
                                valor = intervaloFinal;
                                colunaEsquerda =  colunaEsquerda + (valor - numero);
                                linha = linhaCima+1;
                                coluna = colunaEsquerda+1;
                                System.out.println("Line = "+linha+", column = "+coluna+".");  

                            }
                        }


                    }
                }










                 }

              }
             
            }    
            
            

 }



